File: ApiGateway/src/main/java/ch/emf/youquiz/beans/Question.java
package ch.emf.youquiz.beans;

import java.util.List;

public class Question {

    private Integer pkQuestion;
    private String nom;
    private List<Reponse> reponses;

    public Question(Integer pkQuestion, String nom, List<Reponse> reponses) {
        this.pkQuestion = pkQuestion;
        this.nom = nom;
        this.reponses = reponses;
    }

    // Getters and setters...
    public Integer getPkQuestion() {
        return pkQuestion;
    }

    public void setPkQuestion(Integer pkQuestion) {
        this.pkQuestion = pkQuestion;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public List<Reponse> getReponses() {
        return reponses;
    }

    public void setReponses(List<Reponse> reponses) {
        this.reponses = reponses;
    }
}

File: Client2/js/beans/Question.js
class Question {
    constructor(pkQuestion, nom, reponses) {
        this.pkQuestion = pkQuestion || null;
        this.nom = nom || '';
        this.reponses = reponses || [];
    }

    static fromJSON(questionJSON) {
        let reponses = [];
        if (questionJSON.reponses.length > 0) {
            questionJSON.reponses.forEach(reponseJSON => {
                reponses.push(Reponse.fromJSON(reponseJSON));
            });
        }
        return new Question(questionJSON.pkQuestion, questionJSON.nom, reponses);
    }

    // Getters and setters...
    getPkQuestion() {
        return this.pkQuestion;
    }

    setPkQuestion(pkQuestion) {
        this.pkQuestion = pkQuestion;
    }

    getNom() {
        return this.nom;
    }

    setNom(nom) {
        this.nom = nom;
    }

    getReponses() {
        return this.reponses;
    }

    setReponses(reponses) {
        this.reponses = reponses;
    }
}
